src/archivist/services/index.test.js

```diff
@@ -1,14 +1,506 @@
+
+      context('as a string', () => {
+        it('passes parameters correctly', () => {
+          testParameterPassing('param');
+        });
+      });
+
+      context('as an array', () => {
+        it('passes parameters correctly', () => {
+          testParameterPassing([ 'param1', 'param2' ]);
+        });
+      });
+
+      context('as an object', () => {
+        it('passes parameters correctly', () => {
+          testParameterPassing({ param1: 'param1', param2: 'param2' });
+        });
+      });
+    });
+  });
+
+  describe('#createSourceDocuments', () => {
+    const realFilterNames = Object.keys(exposedFilters);
+    let result;
+    const serviceId = 'service_with_filters_history';
+
+    context('when terms declaration has only one source document', () => {
+      const termsDeclaration = {
+        fetch: 'https://example.com/terms',
+        executeClientScripts: true,
+        select: 'body',
+        remove: '.ads',
+        filter: [
+          realFilterNames[0],
+          'removePrintButton',
+        ],
+      };
+
+      before(async () => {
+        result = await createSourceDocuments(serviceId, termsDeclaration);
+      });
+
+      it('creates a single SourceDocument', () => {
+        expect(result).to.have.length(1);
+        expect(result[0]).to.be.instanceOf(SourceDocument);
+      });
+
+      it('resolves both exposed and custom filters', () => {
+        const sourceDocument = result[0];
+
+        expect(sourceDocument.filters).to.be.an('array');
+        expect(sourceDocument.filters).to.have.length(2);
+        expect(sourceDocument.filters[0]).to.be.a('function');
+        expect(sourceDocument.filters[0]).to.equal(exposedFilters[realFilterNames[0]]);
+        expect(sourceDocument.filters[1]).to.be.a('function');
+        expect(sourceDocument.filters[1].name).to.equal('removePrintButton');
+      });
+
+      it('creates a SourceDocument with the correct properties', () => {
+        const sourceDocument = result[0];
+
+        expect(sourceDocument.location).to.equal(termsDeclaration.fetch);
+        expect(sourceDocument.executeClientScripts).to.equal(termsDeclaration.executeClientScripts);
+        expect(sourceDocument.contentSelectors).to.equal(termsDeclaration.select);
+        expect(sourceDocument.insignificantContentSelectors).to.equal(termsDeclaration.remove);
+      });
+    });
+
+    context('when terms declaration has multiple source documents', () => {
+      const termsDeclaration = {
+        fetch: 'https://example.com/base',
+        executeClientScripts: false,
+        select: 'body',
+        remove: '.base-ads',
+        filter: [
+          realFilterNames[0],
+          'removePrintButton',
+        ],
+        combine: [
+          {
+            fetch: 'https://example.com/doc1',
+            select: '.content',
+            filter: ['removeShareButton'],
+          },
+          {
+            executeClientScripts: true,
+            remove: '.doc2-ads',
+          },
```

Comment:
In [our doc](https://docs.opentermsarchive.org/terms/reference/declaration/), we say:

> Common properties (can be a combination of “select”, “remove”, “filter” and “executeClientScripts”) that are shared across all source documents can be factorized

…yet what I see here is a source document with no `fetch` property. Is that expected to be valid? If so, the doc needs to be corrected.

Reply:
fetch property is factorized and here is `fetch: 'https://example.com/base',`